docs/reference/configuration/configuration.md

```diff
@@ -1,48 +1,276 @@
+
+The configuration system appends a prefix to the name of each parameter, so a parameter with the same name in a library does not conflict with parameters of the same name in targets or other libraries. The prefix is:
+
+| Location | Prefix |
+| -------- | ------ |
+| Target | `target.` |
+| Any library | The name of the library, as found in the `name` section of `mbed_lib.json`, followed by a dot (.) |
+| Application | `app.` |
 
-The default stack size of a user spawned thread is 4 kilobytes. You can configure the stack size of user spawned threads by using the [Mbed config system](/docs/development/tools/index.html#the-configuration-system) or passing in the [Thread constructor](/docs/development/reference/thread.html). You can reduce the default stack size for user spawned threads from 4K to 2K in the following two ways:
+### `mbed_lib.json` format specification
 
-1. Create an `mbed_app.json` file for your project with the following content:
+`mbed_lib.json` is a JSON formatted document that contains a root JSON Object. The keys within this object are sections. See the allowed sections and their meanings below:
 
-    ```JSON
-    {
-        "config": {
-            "thread-stack-size": {
-                "value": 2000
-            }
+| Section | Required | Meaning |
+| ------- | -------- | ------- |
+| `name`  | Yes      | Name of the library. Must be unique. May not be `app` or `target`.|
+| `macros` | No | List of macros to define on the command-line. |
+| `config` | No | Configuration parameters defined for use in this library. |
+| `target_overrides` | No | Overrides for target configuration parameters and configuration parameters of the current library. |
+
+The following is an example library, `mylib`.
+
+```JSON
+{
+    "name": "mylib",
+    "config": {
+        "buffer_size": 1024,
+        "timer_period": {
+            "help": "The timer period (in us)",
+            "macro_name": "INTERNAL_GPTMR_PERIOD",
+            "required": true
+        },
+        "queue_size": {
+            "help": "Size of event queue (entries)",
+            "value": 10
+        }
+    },
+    "macros": ["MYMOD_MACRO1", "MYMOD_MACRO2=\"TEST\""],
+    "target_overrides": {
+        "K64F": {
+             "timer_period": 100,
+             "queue_size": 40
+        },
+        "NXP": {
+             "queue_size": 20,
+             "buffer_size": 128,
+             "target.features_add": ["IPV4"]
         }
     }
-    ```
+}
+```
+
+In this JSON file:
+
+- `name` is the name of the library. **This is a required field.**
+- `config` defines the configuration parameters of the library, as the section about [defining configuration parameters](#defining-configuration-parameters) explains.
+- `macros` is a list of extra macros that are defined when compiling a application that includes this library. `macros` defines the provided list of C preprocessor macros.
```

Comment:
This bullet point is funny, seems like its repeating the same thing.

Also "a application" should be "**an** application".